### DISCUSSION TITLE
Select returns object from db instead of instanciated one when using same type

### DISCUSSION THREAD

**Discussion by ghost** [JasperFx/marten#1314] [Q&A] [answered]
Hi,

I wanted to query objects from the db and create copies of them, with just a few properties set. The goal is to reduce the data serialized. 
Please consider the following classes:

```
public class A
{
  public long Id {get; set;}
  public int A1 {get; set;}
  public int A2 {get; set;}
  public B A3 {get; set;}
}

public class B
{
  //complex implementation ommitted ...
}
```

The query below I expected to yield a list of objects with only A1 and A2 set

```
var aIds = List<long> {1,2,3};        
var reducedA   _documentRepository
                .GetDocuments<A>(a => aIds.contains(a.Id))
                .Select(o => new A { A1 = o.A1, A2 = o.A2 })
                .ToList();
```
But it yielded the original objects stored in the db with all properties set.

Once I change the query to:

```
var aIds = List<long> {1,2,3};        
var reducedA   _documentRepository
                .GetDocuments<A>(a => aIds.contains(a.Id))
                .Select(o => new { o.A1, o.A2 })
                .ToList()
                .Select(o => new A { A1 = o.A1, A2 = o.A2 })
                .ToList();
```

it's working fine and I get objects the have A3 not set.

Regards

**Comment by cocowalla** [comment]
Could you ease include the code for `GetDocuments`, as it's impossible to know exactly what your code is doing otherwise? (You can use a gist if it's too much code for a GitHub comment)

**Comment by ghost** [comment]
sure, sry

```
        public IQueryable<T> GetDocuments<T>(Expression<Func<T, bool>> wherePredicate)
        {
            return _session.Query<T>().Where(wherePredicate);
        }
```

_session is a lightweight session

**Comment by mysticmind** [comment, marked as answer]
@kinl99 Marten currently does not support fetching a subset/projected data using `Select` on the Queryable object but for the whole document. We would possibly look at this use case in 4.x timeframe.
